code/modules/projectiles/guns/energy/laser_gatling.dm

```diff
@@ -115,12 +119,23 @@
 	var/obj/item/minigunpack/ammo_pack
```

Comment:
Ahh, i see what you're doing here

This is bad practice, consider componentizing this instead

Reply:
I guess I could just make it so it detects whether or not it's in a mod suit or not I'll try that

Reply:
Not sure I'm going to start with this I'm really new to this stuff, so I don't really know how to do that because that was just the code there before I started
Is there any examples of something similar like this done correctly?

Reply:
https://hackmd.io/@tgstation/SignalsComponentsElements

Ping me in the code channel if you're not sure after reading this